### DISCUSSION TITLE
Customize JWT

### DISCUSSION THREAD

**Discussion by Immortalin** [directus/directus#4019] [Archive]
Is it possible to customize the JWT and add custom claims?

**Comment by rijkvanzanten** [comment]
No, you can't alter the payload of Directus' access JWT. What would you use it for?

**Reply by Immortalin** [reply]
Claims and permissions for other microservices.

**Reply by naskio** [reply]
It is possible to set a static token for a specific admin user and then use it to make calls from other services.
Doesn't this cover your use case?

more details here: https://docs.directus.io/reference/authentication/#access-tokens

**Comment by development-vargamarcel** [comment]
Would be a nice to have feature. I personally would use it to Authenticate on hasura and use subscriptions until they are available in directus.  @Immortalin

**Reply by rijkvanzanten** [reply]
Interesting use case! Quick question out of personal interest: Isn't Hasura's subscription support a polling-based system, where it effectively makes the same query every second until it detects a change? If that's the case, you might already be able to reproduce that quite easily in Directus

**Reply by development-vargamarcel** [reply]
>  Isn't Hasura's subscription support a polling-based system

you are right,
https://github.com/hasura/graphql-engine/blob/master/architecture/live-queries.md#implementing-graphql-live-queries

But they also make some optimisations: https://hasura.io/docs/latest/graphql/core/databases/postgres/subscriptions/execution-and-performance.html#subscription-multiplexing

> If that's the case, you might already be able to reproduce that quite easily in Directus 

Yeah, you are right again.

**Reply by Keyruu** [reply]
I would like to also integrate Directus with Hasura. Hasura actually is a nice way to unify all your GraphQL endpoints in one point.

**Reply by alexdabast** [reply]
Would be nice indeed to have a way to customize the jwt we want to use it in order to log our users in other system like ejabbered

**Reply by azrikahar** [reply]
@alexdabast you should be able to customize it as it's already implemented in the form of [`auth.jwt` filter hook event](https://docs.directus.io/extensions/hooks.html#filter-events)! 👍

Just as further reference, it was tracked as a feature request in #5406 and subsequently implemented in #7906 on `v9.0.0-rc.94`.
